integrations/pgvector/src/haystack_integrations/document_stores/pgvector/document_store.py

```diff
@@ -112,6 +119,12 @@ def __init__(
             `SELECT cfgname FROM pg_ts_config;`.
             More information can be found in this [StackOverflow answer](https://stackoverflow.com/a/39752553).
         :param embedding_dimension: The dimension of the embedding.
+        :param vector_type: The type of the vector to use for the embedding.
+            `"vector"` is the default and uses the `vector` type in PostgreSQL.
+            `"halfvec"` uses the `halfvec` type in PostgreSQL, which uses half the memory but less precision.
+            `"halfvec"` is only supported for pg_vector versions 0.7.0 and above.
+            Use `"halfvec"` if your need to support more than 2,000 dimensions for embeddings.
+            More information can be found in the [Pgvector issue](https://github.com/pgvector/pgvector/issues/461).
```

Comment:
I would link this README section: https://github.com/pgvector/pgvector?tab=readme-ov-file#what-if-i-want-to-index-vectors-with-more-than-2000-dimensions